solidity/test/token/HypERC20CollateralVaultDeposit.t.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.13;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+import "forge-std/Test.sol";
+import {ERC4626Test} from "../../contracts/test/ERC4626/ERC4626Test.sol";
+import {TypeCasts} from "../../contracts/libs/TypeCasts.sol";
+import {HypTokenTest} from "./HypERC20.t.sol";
+
+import {HypERC20CollateralVaultDeposit} from "../../contracts/token/HypERC20CollateralVaultDeposit.sol";
+import "../../contracts/test/ERC4626/ERC4626Test.sol";
+
+contract HypERC20CollateralVaultDepositTest is HypTokenTest {
+    using TypeCasts for address;
+    uint constant DUST_AMOUNT = 1e11;
+    HypERC20CollateralVaultDeposit internal erc20CollateralVaultDeposit;
+    ERC4626Test vault;
+
+    function setUp() public override {
+        super.setUp();
+        vault = new ERC4626Test(address(primaryToken), "Regular Vault", "RV");
+
+        localToken = new HypERC20CollateralVaultDeposit(
+            address(vault),
+            address(primaryToken),
+            address(localMailbox)
+        );
+        erc20CollateralVaultDeposit = HypERC20CollateralVaultDeposit(
+            address(localToken)
+        );
```

Comment:
why is this wrapped twice

Reply:
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/d3d7ad7f973e52b60e9119cc98e82bca9a53f707/solidity/test/token/HypERC20.t.sol#L177-L178

Something similar is done in this test. If i understand correctly, it is used to set the localToken in the child class that is a `TokenRouter` type.